core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -0,0 +1,154 @@
+use super::broadcast_utils;
+use super::*;
+use crate::entry::EntrySlice;
+use rayon::prelude::*;
+use solana_sdk::signature::Signable;
+
+#[derive(Default)]
+struct BroadcastStats {
+    num_entries: Vec<usize>,
+    run_elapsed: Vec<u64>,
+    to_blobs_elapsed: Vec<u64>,
+}
+
+pub(super) struct StandardBroadcastRun {
+    stats: BroadcastStats,
+}
+
+impl StandardBroadcastRun {
+    pub(super) fn new() -> Self {
+        Self {
+            stats: BroadcastStats::default(),
+        }
+    }
+
+    fn update_broadcast_stats(
+        &mut self,
+        broadcast_elapsed: u64,
+        run_elapsed: u64,
+        num_entries: usize,
+        to_blobs_elapsed: u64,
+        blob_index: u64,
+    ) {
+        inc_new_counter_info!("broadcast_service-time_ms", broadcast_elapsed as usize);
+
+        self.stats.num_entries.push(num_entries);
+        self.stats.to_blobs_elapsed.push(to_blobs_elapsed);
+        self.stats.run_elapsed.push(run_elapsed);
+        if self.stats.num_entries.len() >= 16 {
+            info!(
+                "broadcast: entries: {:?} blob times ms: {:?} broadcast times ms: {:?}",
+                self.stats.num_entries, self.stats.to_blobs_elapsed, self.stats.run_elapsed
+            );
+            self.stats.num_entries.clear();
+            self.stats.to_blobs_elapsed.clear();
+            self.stats.run_elapsed.clear();
+        }
+
+        datapoint!("broadcast-service", ("transmit-index", blob_index, i64));
+    }
+}
+
+impl BroadcastRun for StandardBroadcastRun {
```

Comment:
this is a ton of duplicated code for the one-line delta...

Reply:
agreed, factored out the recv_blobs code but didn't do the rest. Will pull the rest of this into utils as well.